Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: qiskit-aer-gpu in /home/gharrabi/.local/lib/python3.10/site-packages (0.14.2)
Requirement already satisfied: cuquantum-cu12>=23.3.0 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (24.3.0.post1)
Requirement already satisfied: psutil>=5 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (5.9.8)
Requirement already satisfied: nvidia-cusparse-cu12>=12.1.0.106 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (12.4.1.24)
Requirement already satisfied: scipy>=1.0 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (1.13.1)
Requirement already satisfied: nvidia-cublas-cu12>=12.1.3.1 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (12.5.2.13)
Requirement already satisfied: nvidia-cusolver-cu12>=11.4.5.107 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (11.6.2.40)
Requirement already satisfied: qiskit>=0.45.2 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (1.1.0)
Requirement already satisfied: nvidia-nvjitlink-cu12 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (12.5.40)
Requirement already satisfied: nvidia-cuda-runtime-cu12>=12.1.105 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (12.5.39)
Requirement already satisfied: numpy>=1.16.3 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit-aer-gpu) (1.26.4)
Requirement already satisfied: custatevec-cu12==1.6.0.post1 in /home/gharrabi/.local/lib/python3.10/site-packages (from cuquantum-cu12>=23.3.0->qiskit-aer-gpu) (1.6.0.post1)
Requirement already satisfied: cutensornet-cu12==2.4.0.post1 in /home/gharrabi/.local/lib/python3.10/site-packages (from cuquantum-cu12>=23.3.0->qiskit-aer-gpu) (2.4.0.post1)
Requirement already satisfied: cutensor-cu12<3,>=2.0.1 in /home/gharrabi/.local/lib/python3.10/site-packages (from cutensornet-cu12==2.4.0.post1->cuquantum-cu12>=23.3.0->qiskit-aer-gpu) (2.0.1)
Requirement already satisfied: sympy>=1.3 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit>=0.45.2->qiskit-aer-gpu) (1.12.1)
Requirement already satisfied: stevedore>=3.0.0 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit>=0.45.2->qiskit-aer-gpu) (5.2.0)
Requirement already satisfied: dill>=0.3 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit>=0.45.2->qiskit-aer-gpu) (0.3.8)
Requirement already satisfied: typing-extensions in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit>=0.45.2->qiskit-aer-gpu) (4.12.2)
Requirement already satisfied: symengine>=0.11 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit>=0.45.2->qiskit-aer-gpu) (0.11.0)
Requirement already satisfied: python-dateutil>=2.8.0 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit>=0.45.2->qiskit-aer-gpu) (2.9.0.post0)
Requirement already satisfied: rustworkx>=0.14.0 in /home/gharrabi/.local/lib/python3.10/site-packages (from qiskit>=0.45.2->qiskit-aer-gpu) (0.14.2)
Requirement already satisfied: six>=1.5 in /home/gharrabi/.local/lib/python3.10/site-packages (from python-dateutil>=2.8.0->qiskit>=0.45.2->qiskit-aer-gpu) (1.16.0)
Requirement already satisfied: pbr!=2.1.0,>=2.0.0 in /home/gharrabi/.local/lib/python3.10/site-packages (from stevedore>=3.0.0->qiskit>=0.45.2->qiskit-aer-gpu) (6.0.0)
Requirement already satisfied: mpmath<1.4.0,>=1.1.0 in /home/gharrabi/.local/lib/python3.10/site-packages (from sympy>=1.3->qiskit>=0.45.2->qiskit-aer-gpu) (1.3.0)
           ┌───┐           ┌───┐        ┌───┐   ┌───┐         ░ ┌─┐         
   q_0: ───┤ H ├─────■─────┤ H ├────────┤ X ├───┤ H ├─────■───░─┤M├─────────
        ┌──┴───┴──┐┌─┴─┐┌──┴───┴──┐     └─┬─┘┌──┴───┴──┐┌─┴─┐ ░ └╥┘┌─┐      
   q_1: ┤ Rx(0.5) ├┤ X ├┤ Rx(0.5) ├──■────┼──┤ Rx(0.5) ├┤ X ├─░──╫─┤M├──────
        ├─────────┤└───┘├─────────┤┌─┴─┐  │  ├─────────┤└───┘ ░  ║ └╥┘┌─┐   
   q_2: ┤ Rz(1.2) ├──■──┤ Rz(1.2) ├┤ X ├──┼──┤ Rz(1.2) ├──■───░──╫──╫─┤M├───
        ├─────────┤┌─┴─┐├─────────┤└───┘  │  ├─────────┤┌─┴─┐ ░  ║  ║ └╥┘┌─┐
   q_3: ┤ Ry(0.3) ├┤ X ├┤ Ry(0.3) ├───────■──┤ Ry(0.3) ├┤ X ├─░──╫──╫──╫─┤M├
        └─────────┘└───┘└─────────┘          └─────────┘└───┘ ░  ║  ║  ║ └╥┘
meas: 4/═════════════════════════════════════════════════════════╩══╩══╩══╩═
                                                                 0  1  2  3 
Traceback (most recent call last):
  File "/work/ccresearch/gharrabi/expr/exp5/exprz/Four-Qubit3.py", line 48, in <module>
    result = simulator.run(qc, shots=1024).result()
  File "/home/gharrabi/.local/lib/python3.10/site-packages/qiskit_aer/jobs/utils.py", line 42, in _wrapper
    return func(self, *args, **kwargs)
  File "/home/gharrabi/.local/lib/python3.10/site-packages/qiskit_aer/jobs/aerjob.py", line 114, in result
    return self._future.result(timeout=timeout)
  File "/opt/rit/el9/20230413/app/linux-rhel9-x86_64_v3/gcc-11.2.1/miniconda3-22.11.1-hydt3qzqzg5alf7csvqt24mf7xt64o53/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/opt/rit/el9/20230413/app/linux-rhel9-x86_64_v3/gcc-11.2.1/miniconda3-22.11.1-hydt3qzqzg5alf7csvqt24mf7xt64o53/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/opt/rit/el9/20230413/app/linux-rhel9-x86_64_v3/gcc-11.2.1/miniconda3-22.11.1-hydt3qzqzg5alf7csvqt24mf7xt64o53/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/gharrabi/.local/lib/python3.10/site-packages/qiskit_aer/backends/aerbackend.py", line 474, in _execute_circuits_job
    output = self._execute_circuits(aer_circuits, noise_model, config)
  File "/home/gharrabi/.local/lib/python3.10/site-packages/qiskit_aer/backends/aer_simulator.py", line 923, in _execute_circuits
    ret = cpp_execute_circuits(self._controller, aer_circuits, noise_model, config)
  File "/home/gharrabi/.local/lib/python3.10/site-packages/qiskit_aer/backends/backend_utils.py", line 450, in cpp_execute_circuits
    return controller.execute(aer_circuits, noise_model, config)
RuntimeError: No CUDA device available!
